lib/final-api/v1/http/ddtf_build.rb

```diff
@@ -113,6 +125,22 @@ def parts_status
           end
         end
 
+        def map_step_result_state
+          {
+            'created' => 'NotSet',
+            'blocked' => 'NotTested',
+            'passed' => 'Passed',
+            'failed' => 'Failed'
+            # 'pending' => is handled elsewhere
+          }
+        end
+
+        def resolve_step_result(old_step_result)
+          old_step_result[:result].downcase == 'pending' ?
+            (old_step_result[:data][:status].to_s.downcase == 'not_performed'? 'NotPerformed' : 'Skipped') :
+            map_step_result_state[old_step_result[:result]]
```

Comment:
use this: `state2api_v1status(old_step_result[:data][:status] || old_step_result[:result])`

it will not work until bug with pending status will be fixed.